src/Microsoft.ML.Data/DataLoadSave/Binary/BinaryLoader.cs

```diff
@@ -2037,7 +2044,7 @@ protected override bool MoveNextCore()
                         // threads will exit if all potentially blocking operations are
                         // waiting on the same cancellation token that we catch here.
                         Contracts.Assert(ex.CancellationToken == _exMarshaller.Token);
-                        _disposed = true;
```

Comment:
>_disposed = true; [](start = 24, length = 17)

I notice we've removed `_disposed` but not replaced it with a `_done`. I *think* it should also be a `_done` here, since that flag controls whether or not the dispose method will try to clean up the threads, and the comment above seems to indicate that in such a case as this it is not necessary to clean up the threads, so we should still set the `_done` flag that suppresses the thread cleanup? Is it not so?

Reply:
Right. Thanks, I wasn't sure about this case :-).

---
In reply to: [243345481](https://github.com/dotnet/machinelearning/pull/1928#discussion_r243345481) [](ancestors = 243345481)